test/src/e2e_vm_tests/test_programs/should_pass/stdlib/vec/src/main.sw

```diff
@@ -0,0 +1,719 @@
+script;
+
+use std::{assert::assert, hash::sha256, option::Option, revert::revert, vec::Vec};
+
+struct S {
+    x: u32,
+    y: b256,
+}
+
+enum E {
+    X: (),
+    Y: b256,
+    Z: (b256,
+    b256), 
+}
+
+fn main() -> bool {
+    test_vector_new_u8();
+    test_vector_new_b256();
+    test_vector_new_struct();
+    test_vector_new_enum();
+    test_vector_new_tuple();
+    test_vector_new_string();
+    test_vector_new_array();
+    test_vector_with_capacity_u64();
+    true
+}
+
+fn test_vector_new_u8() {
+    let mut v: Vec<u8> = ~Vec::new::<u8>();
+
+    let n0 = 0u8;
+    let n1 = 1u8;
+    let n2 = 2u8;
+    let n3 = 3u8;
+    let n4 = 4u8;
+    let n5 = 5u8;
+    let n6 = 6u8;
+    let n7 = 7u8;
+    let n8 = 8u8;
+
+    assert(v.len() == 0);
+    assert(v.capacity() == 0);
+    assert(v.is_empty() == true);
+
+    v.push(n0);
+    v.push(n1);
+    v.push(n2);
+    v.push(n3);
+    v.push(n4);
+
+    assert(v.len() == 5);
+    assert(v.capacity() == 8);
+    assert(v.is_empty() == false);
+
+    match v.get(0) {
+        Option::Some(val) => assert(val == n0), Option::None => revert(0), 
+    }
+
+    // Push after get
+    v.push(n5);
+    v.push(n6);
+    v.push(n7);
+    v.push(n8);
+
+    match v.get(4) {
+        Option::Some(val) => assert(val == n4), Option::None => revert(0), 
+    }
+
+    match v.get(n6) {
+        Option::Some(val) => assert(val == n6), Option::None => revert(0), 
+    }
+
+    assert(v.len() == 9);
+    assert(v.capacity() == 16);
+    assert(v.is_empty() == false);
+
+    // Test after capacity change
+    match v.get(4) {
+        Option::Some(val) => assert(val == n4), Option::None => revert(0), 
+    }
+
+    match v.get(6) {
+        Option::Some(val) => assert(val == n6), Option::None => revert(0), 
+    }
+
+    v.clear();
+
+    // Empty after clear
+    assert(v.len() == 0);
+    assert(v.capacity() == 16);
```

Comment:
So, this might be standard behaviour for a vec (I'm just ignorant) however wouldn't this cause eventual bloat on the chain? Ofc, zeroing out the data is not something we want to do since marking it as empty should be good enough to have something else overwrite it but if the vec continues to grow in capacity would that cause an issue somewhere?
Why would we not nuke the capacity too?

Reply:
This is in memory, not state. Memory is cleared once the transaction execution completes.